app/src/main/java/mozilla/lockbox/presenter/AutofillRoutePresenter.kt

```diff
@@ -82,10 +96,16 @@ class AutofillRoutePresenter(
 
     private fun route(action: RouteAction) {
         when (action) {
-            is RouteAction.LockScreen -> navigateToFragment(R.id.fragment_locked)
-            is RouteAction.ItemList -> showDialogFragment(AutofillFilterFragment(),
-                RouteAction.DialogFragment.AutofillSearchDialog
-            )
+            is RouteAction.LockScreen -> {
+                dismissDialogIfPresent()
+                navigateToFragment(R.id.fragment_locked)
+            }
+            is RouteAction.ItemList -> {
+                dismissDialogIfPresent()
```

Comment:
I'm still not altogether happy with `dismissDialogIfPresent`. It feels like the dialog is not cleaning up after its own mess.

Reply:
Can this be handled in the navigation instead of here? It doesn't seem to make sense.